### DISCUSSION TITLE
Does this support Proto Datastore?

### DISCUSSION THREAD

**Discussion by kawpii** [russhwolf/multiplatform-settings#105] [Q&A] [answered]
https://developer.android.com/topic/libraries/architecture/datastore#datastore-typed comes with type safety

**Comment by russhwolf** [comment, marked as answer]
No. You can use preference datastore, but in order to keep consistent with other platform APIs the common API is based on string keys.

The library does provide delegates which can be helpful for type-safety
```kotlin
val settings: Settings = //...
val someData: String by settings.string("key", defaultValue = "")
```
